src/mempool.cpp

```diff
@@ -132,7 +131,20 @@ PopData MemPool::getPop(AltTree& tree) {
   return ret;
 }
 
-void MemPool::removePayloads(const PopData& pop) {
+void MemPool::filterVbkBlocks(const AltTree& tree) {
+  for (auto rel_it = relations_.begin(); rel_it != relations_.end();) {
+    if (tree.vbk().getBlockIndex(rel_it->second->header->getHash()) !=
```

Comment:
auto blockHash = rel_it->second->header->getHash();
auto* index = tree.vbk().getBlockIndex(blockHash);
if(index != nullptr && rel_it->second->empty()) {
...
}

Reply:
(to improve readability)

Reply:
Fixed